lib/entice/logic/vitals.ex

```diff
@@ -21,44 +23,97 @@ defmodule Entice.Logic.Vitals do
     alias Entice.Logic.Vitals.Health
     alias Entice.Logic.Vitals.Energy
+    alias Entice.Logic.Vitals.Morale
     alias Entice.Logic.Player.Level
 
 
     def init(entity, _args) do
       {:ok, entity |> put_attribute(get_max_health(entity))
-                   |> put_attribute(get_max_energy(entity))}
+                   |> put_attribute(get_max_energy(entity))
+                   |> put_attribute(%Morale{morale: 0})}
+    end
 
+    def init(entity, [:entity_resurrected, percent_health, percent_energy]) do
+      {_, max_health: max_health} = get_max_health(entity)
+      resurrected_health = max_health / 100 * percent_health
+
+      {_, max_mana: max_mana} = get_max_energy(entity)
+      resurrected_mana = max_mana / 100 * percent_energy
+
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: resurrected_health, max_health: max_health} end)
+                   |> update_attribute(Energy, fn _ -> %Energy{mana: resurrected_mana, max_mana: max_mana} end)}
     end
 
+    def termiante(:remove_handler, entity) do
+      {:ok, entity}
+    end
 
     def terminate(_reason, entity) do
       {:ok, entity |> remove_attribute(Health)
                    |> remove_attribute(Energy)}
     end
 
+    def handle_event({:vitals_entity_damage, amount}, %Entity{id: id, attributes: %{Health => %Health{health: health, max_health: max_health}}} = entity) do
+      new_health = health - amount
+      cond do
+        new_health <= 0 -> {:become, Vitals.DeadBehaviour, [], entity |> update_attribute(Health, fn health -> %Health{health: 0, max_health: health.max_health} end)}
+        new_health > 0 -> {:ok, entity |> update_attribute(Health, fn health -> %Health{health: new_health, max_health: health.max_health} end)}
+      end
+    end
 
-    # internal
+    def handle_event({:vitals_entity_heal, amount}, %Entity{id: id} = entity) do
+      {:ok, %Health{health: health, max_health: max_health}} = fetch_attribute(entity, Health)
+      new_health = health + ammount
+      cond do
+        (health + amount) > max_health -> new_health = max_health
+        (health + amount) < max_health -> new_health = health + amount
+      end
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: new_health, max_health: max_health})}
+    end
 
+    # internal
 
     defp get_max_health(entity) do
       {:ok, level} = fetch_attribute(entity, Level)
+      {:ok, morale} = fetch_attribute(entity, Morale)
       health = calc_life_points_for_level(level.level)
+      max_health_with_morale = health / 100 * (100 + morale.morale)
       %Health{health: health, max_health: health}
     end
 
-
     #TODO: Take care of Armor, Runes, Weapons...
     defp calc_life_points_for_level(level),
     do: 100 + ((level - 1) * 20) # Dont add 20 lifePoints for level1
 
 
     #TODO: Take care of Armor, Runes, Weapons...
-    defp get_max_energy(_entity),
-    do: %Energy{mana: 70, max_mana: 70}
+    defp get_max_energy(_entity) do
```

Comment:
jeah thats right. but we also have to calculate the level depending on the class i think.
But i will add energy depending on morale in this MS